src/common/io/socket/common.c

```diff
@@ -129,6 +141,52 @@ sckOptionSet(int fd)
     FUNCTION_TEST_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+static bool
+sckConnectInProgress(int errNo)
+{
+    return errNo == EINPROGRESS || errNo == EINTR;
+}
+
+void
+sckConnect(int fd, const String *host, unsigned int port, const struct addrinfo *hostAddress, TimeMSec timeout)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(INT, fd);
+        FUNCTION_LOG_PARAM(STRING, host);
+        FUNCTION_LOG_PARAM(UINT, port);
+        FUNCTION_LOG_PARAM_P(VOID, hostAddress);
+        FUNCTION_LOG_PARAM(TIME_MSEC, timeout);
+    FUNCTION_LOG_END();
+
+    // Attempt connection
+    if (connect(fd, hostAddress->ai_addr, hostAddress->ai_addrlen) == -1)
+    {
+        // Save the error
+        int errNo = errno;
+
+        // The connection has started but since we are in non-blocking mode it has not completed yet
+        if (sckConnectInProgress(errNo))
+        {
+            // Wait for write-ready
+            if (!sckReadyWrite(fd, timeout))
+                THROW_FMT(HostConnectError, "timeout connecting to '%s:%u'", strPtr(host), port);
+
+            // Check that the connection was successful
+            socklen_t errNoLen = sizeof(errNo);
+
+            THROW_ON_SYS_ERROR(
+                getsockopt(fd, SOL_SOCKET, SO_ERROR, &errNo, &errNoLen) == -1, HostConnectError, "unable to get socket error");
```

Comment:
So here you are calling getsockopt with SO_ERROR to actually check for and clear an error at the same time? When I first read trough this section of the code, I thought we were always erroring because we are getting SO_ERROR. Maybe just a comment to say "call getsockopt with SO_ERROR to clear any errors if the connection was successful or report on an unsucessful connection" - or something. Dunno. Your choice. It's clever but it just took me a bit to figure out what was going on here.

Reply:
Yeah, I can add to the comment, but we don't want two error codes because that makes to code overall more complicated.

Reply:
Changed the comment to `Check for success or error. If the connection was successful this will set errNo to 0.`